compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -582,6 +596,68 @@ fn report_missing_placeholders(
         })
     };
 
+    let placeholders = pieces
```

Comment:
Could you move this to a function which adds to the diagnostic? Just so `report_missing_placeholder` doesn't get too large?